### PR TITLE
refactor(accordion): new proposed api and implementation

### PR DESCRIPTION
**Pull request by thatsamsonkid**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?
Primarily non-breaking refactor with the exception of the accordion trigger

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [x] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] button
- [ ] calendar
- [ ] card
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [ ] hover-card
- [ ] input
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] tooltip
- [ ] typography

## What is the current behavior?

Currently accordion trigger has been kept as a directive due to issues with 'class' being applied to both host element and underlying template element. However I believe it would be better suited as a component and but while current impl is flexible it also allows for misuse even at hlm level. Also from a usage standpoint the flipping between having to provide component selector vs directive on an element is not necessarily the best DX or at least not predictable.

## What is the new behavior?
This PR switches the trigger to directive and also fixes that initial issue related to class when initially tried to be used as a component.

Also this pr updates accordion-content implementation to allow the brn component to also assume the hlm selector in order not to duplicate any logic required for functionality but also paired it with a hlm directive that also targets hlm-accordion-content selector in order to apply the required classes in hlm. Also added an opt out for the directive as well to avoid any potential conflicts

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

Currently accordion trigger we need to provide the button and attach the directive but going forward hlm and brn trigger will provide the underlying button. Ensures proper implementation regardless of whether using hlm or brn. Most of the existing usage will continue to work with he exception of the trigger

```
    <button hlmAccordionTrigger>
      Is it accessible?
      <hlm-icon hlmAccIcon />
    </button>
```
to 
```
      <hlm-accordion-trigger>
        Is it accessible?
        <hlm-icon hlmAccIcon />
      </hlm-accordion-trigger>
```

## Other information

Overall api changes for accordion in this PR changes from  
```
<div hlmAccordion>
  <div hlmAccordionItem>
    <button hlmAccordionTrigger>
      Is it accessible?
      <hlm-icon hlmAccIcon />
    </button>
    <brn-accordion-content hlm>Yes. It adheres to the WAI-ARIA design pattern.</brn-accordion-content>
  </div>
  </div>
```
moves to 
```
  <hlm-accordion>
    <hlm-accordion-item>
      <hlm-accordion-trigger>
        Is it accessible?
        <hlm-icon hlmAccIcon />
      </hlm-accordion-trigger>
      <hlm-accordion-content>
        Yes. It adheres to the WAI-ARIA design pattern.
      </hlm-accordion-content>
    </hlm-accordion-item>
  <hlm-accordion>
```
From hlm perspective this looks a little more consistent and comparable to what is done on shadcn/radix. Its a little more intuitive/predictable and still customizable. Hoping maybe this strikes a better balance in terms of api, flexbility and ensuring proper usage. 

There's is also a question of whether a heading tag should be added around the trigger which this implementation would allow us to do pretty easily if we decide that is the best direction to go in and would align more closely with shadcn/radix

### CODE REVIEW

**Comment by thatsamsonkid** [comment]
Just a heads up I still need to update the tests but wanted to get some eyes on this before going ahead and doing that but did some smoke testing and all seems good so far. I'm also again as we talked about before good with leaving the trigger as is or if you want we can split off all the other changes and maybe give the accordion-trigger a little more time and thought to ensure we cover all use cases there. Don't want to rush a conclusion on that one.

File: libs/cli/src/generators/ui/libs/ui-accordion-helm/files/lib/hlm-accordion-trigger.directive.ts.template [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { ClassValue } from 'clsx';
 		'[style.--tw-ring-offset-shadow]': '"0 0 #000"',
 		'[class]': '_computedClass()',
 	},
-	hostDirectives: [BrnAccordionTriggerDirective],
+	hostDirectives: [BrnAccordionTriggerComponent],
```
Comment (ajitzero) [inline libs/cli/src/generators/ui/libs/ui-accordion-helm/files/lib/hlm-accordion-trigger.directive.ts.template:13]: Question: Is this allowed in Angular? Using a component as a directive?

Comment (thatsamsonkid) [reply libs/cli/src/generators/ui/libs/ui-accordion-helm/files/lib/hlm-accordion-trigger.directive.ts.template:13]: No and good catch! I will need to remove that. Got carried away with find and replace haha

File: libs/ui/accordion/brain/src/lib/brn-accordion-trigger.component.ts [hunk 2]
```diff
@@ -38,15 +63,22 @@ export class BrnAccordionTriggerDirective {
...
 		});
+	}
 
-		fromEvent(this._elementRef.nativeElement, 'focus')
-			.pipe(takeUntilDestroyed())
-			.subscribe(() => {
-				this._accordion.setActiveItem(this);
-			});
+	ngOnInit(): void {
+		fromEvent(this.trigger.nativeElement, 'focus')
```
Comment (gergobergo) [inline libs/ui/accordion/brain/src/lib/brn-accordion-trigger.component.ts:69]: question: how do you handle the unsubscribe from the `fromEvent` Observable?

Comment (goetzrobin) [reply libs/ui/accordion/brain/src/lib/brn-accordion-trigger.component.ts:69]: The takeUntilDestroyed operator takes care of that. More here: https://angular.io/api/core/rxjs-interop/takeUntilDestroyed

Comment (gergobergo) [reply libs/ui/accordion/brain/src/lib/brn-accordion-trigger.component.ts:69]: Crap, I just missed that line. :)

**Comment by goetzrobin** [comment]
@thatsamsonkid sorry. I meant to open a PR to your branch and accidentally pushed my changes in a commit... Can you let me know if this looks good? Love this PR and the new API surface! We can then adjust the tests and get this merged ASAP! I think we are finding some great patterns to work around the "Angular limitations" and I am getting more and more excited about this every day!

**Comment by goetzrobin** [comment]
Can we add this fix from @elite-benni, @thatsamsonkid?

**Comment by thatsamsonkid** [comment]
Sure thing!

**Comment by goetzrobin** [comment]
@thatsamsonkid I can take a look at this again if you don't have the bandwidth due to your hard work on the `select`! Let me know how I can help!

**Comment by thatsamsonkid** [comment]
> @thatsamsonkid I can take a look at this again if you don't have the bandwidth due to your hard work on the `select`! Let me know how I can help!

I made the changes last night. I just didn't get push them up yet 😭. Ill push them tonight and you can review

**Comment by goetzrobin** [comment]
@thatsamsonkid awesome stuff! Thanks so much!!

**Comment by thatsamsonkid** [comment]
Continuing with the none breaking changes DX improvements for accordion to PR (#126). Unfortunately ran into an issue trying to convert the trigger to a component in both Hlm or Brn so that would need some additional time to look into but for not much gain I feel at the moment. So moved all none trigger changes to above PR and closing this one for now.

**Pull request closed without merging**